api/checks/suites.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package checks
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"google.golang.org/appengine/datastore"
+)
+
+func getOrCreateCheckSuite(ctx context.Context, sha, owner, repo string, installation int64) (*shared.CheckSuite, error) {
+	query := datastore.NewQuery("CheckSuite").
+		Filter("SHA =", sha).
+		Filter("InstallationID =", installation).
+		Filter("Owner =", owner).
+		Filter("Repo =", repo).
+		KeysOnly()
+	var suite shared.CheckSuite
+	if keys, err := query.GetAll(ctx, nil); err != nil {
+		return nil, err
+	} else if len(keys) > 0 {
+		err := datastore.Get(ctx, keys[0], &suite)
+		return &suite, err
+	}
+
+	log := shared.GetLogger(ctx)
+	suite.SHA = sha
+	suite.Owner = owner
+	suite.Repo = repo
+	suite.InstallationID = installation
+	_, err := datastore.Put(ctx, datastore.NewIncompleteKey(ctx, "CheckSuite", nil), &suite)
+	if err != nil {
+		log.Debugf("Created CheckSuite entity for %s/%s @ %s", owner, repo, sha)
+	}
+	return &suite, err
+}
+
+// GetSuitesForSHA returns any CheckSuite entities associated with the given SHA.
+func GetSuitesForSHA(ctx context.Context, sha string) ([]shared.CheckSuite, error) {
+	var suites []shared.CheckSuite
+	_, err := datastore.NewQuery("CheckSuite").Filter("SHA =", sha).GetAll(ctx, &suites)
+	return suites, err
+}
+
+// PendingCheckRun loads the CheckSuite(s), if any, for the given SHAs, and creates a pending
+// check_run for the given browser name for each CheckSuite.
```

Comment:
Clarify the return value (i.e. whether a new run is created; also explain when it will create a run).

Reply:
Done.